File: kafkaStreamsModelServer/src/main/java/com/lightbend/java/kafkastreams/modelserver/memorystore/PrintProcessor.java
package com.lightbend.java.kafkastreams.modelserver.memorystore;

import com.lightbend.java.model.ServingResult;
import org.apache.kafka.streams.processor.AbstractProcessor;

/**
 * Created by boris on 7/12/17.
 */

public class PrintProcessor extends AbstractProcessor<byte[], ServingResult> {

    @Override
    public void process(byte[] key, ServingResult value) {
        if(value.isProcessed())
            System.out.println("Calculated quality - " + value.getResult() + " in " + value.getDuration() + "ms");
        else
            System.out.println("No model available - skipping");
    }
}

File: kafkaStreamsModelServer/src/main/scala/com/lightbend/scala/kafkastreams/modelserver/memorystore/PrintProcessor.scala
package com.lightbend.scala.kafkastreams.modelserver.memorystore

import com.lightbend.scala.modelServer.model.ServingResult
import org.apache.kafka.streams.processor.AbstractProcessor

/**
 * Implements a topology processor that just prints out the results to stdout.
 */
class PrintProcessor extends AbstractProcessor[Array[Byte], ServingResult]{

  override def process (key: Array[Byte], value: ServingResult ): Unit = {
    value.processed match {
      case true => println(s"Calculated quality - ${value.result} calculated in ${value.duration} ms")
      case _ => println("No model available - skipping")
    }
  }
}
